tests/integration/bank/keeper/keeper_test.go

```diff
@@ -0,0 +1,1667 @@
+	"github.com/cosmos/cosmos-sdk/x/bank/exported"
+	"github.com/cosmos/cosmos-sdk/x/bank/keeper"
+	"github.com/cosmos/cosmos-sdk/x/bank/testutil"
+	"github.com/cosmos/cosmos-sdk/x/bank/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	minttypes "github.com/cosmos/cosmos-sdk/x/mint/types"
+)
+
+const (
+	fooDenom     = "foo"
+	barDenom     = "bar"
+	initialPower = int64(100)
+	holder       = "holder"
+	multiPerm    = "multiple permissions account"
+	randomPerm   = "random permission"
+)
+
+var (
+	holderAcc     = authtypes.NewEmptyModuleAccount(holder)
+	burnerAcc     = authtypes.NewEmptyModuleAccount(authtypes.Burner, authtypes.Burner)
+	minterAcc     = authtypes.NewEmptyModuleAccount(authtypes.Minter, authtypes.Minter)
+	multiPermAcc  = authtypes.NewEmptyModuleAccount(multiPerm, authtypes.Burner, authtypes.Minter, authtypes.Staking)
+	randomPermAcc = authtypes.NewEmptyModuleAccount(randomPerm, "random")
+
+	// The default power validators are initialized to have within tests
+	initTokens = sdk.TokensFromConsensusPower(initialPower, sdk.DefaultPowerReduction)
+	initCoins  = sdk.NewCoins(sdk.NewCoin(sdk.DefaultBondDenom, initTokens))
+)
+
+func newFooCoin(amt int64) sdk.Coin {
+	return sdk.NewInt64Coin(fooDenom, amt)
+}
+
+func newBarCoin(amt int64) sdk.Coin {
+	return sdk.NewInt64Coin(barDenom, amt)
+}
+
+// nolint: interfacer
+func getCoinsByName(ctx sdk.Context, bk keeper.Keeper, ak types.AccountKeeper, moduleName string) sdk.Coins {
+	moduleAddress := ak.GetModuleAddress(moduleName)
+	macc := ak.GetAccount(ctx, moduleAddress)
+	if macc == nil {
+		return sdk.Coins(nil)
+	}
+
+	return bk.GetAllBalances(ctx, macc.GetAddress())
+}
+
+type IntegrationTestSuite struct {
+	suite.Suite
+
+	app         *simapp.SimApp
+	ctx         sdk.Context
+	queryClient types.QueryClient
+	msgServer   types.MsgServer
+}
+
+func (suite *IntegrationTestSuite) initKeepersWithmAccPerms(blockedAddrs map[string]bool) (authkeeper.AccountKeeper, keeper.BaseKeeper) {
+	app := suite.app
+	maccPerms := simapp.GetMaccPerms()
+	appCodec := moduletestutil.MakeTestEncodingConfig(auth.AppModuleBasic{}, bank.AppModuleBasic{}).Codec
+
+	maccPerms[holder] = nil
+	maccPerms[authtypes.Burner] = []string{authtypes.Burner}
+	maccPerms[authtypes.Minter] = []string{authtypes.Minter}
+	maccPerms[multiPerm] = []string{authtypes.Burner, authtypes.Minter, authtypes.Staking}
+	maccPerms[randomPerm] = []string{"random"}
+	authKeeper := authkeeper.NewAccountKeeper(
+		appCodec, app.GetKey(types.StoreKey), authtypes.ProtoBaseAccount,
+		maccPerms, sdk.Bech32MainPrefix, authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+	)
+	keeper := keeper.NewBaseKeeper(
+		appCodec, app.GetKey(types.StoreKey), authKeeper, blockedAddrs, authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+	)
+
+	return authKeeper, keeper
+}
+
+func (suite *IntegrationTestSuite) SetupTest() {
+	suite.app = simapp.Setup(suite.T(), false)
```

Comment:
From https://github.com/cosmos/cosmos-sdk/pull/12789, we should use depinject instead of `simapp` ?

Reply:
Or maybe it's fine to use simapp in `tests/integration`

Reply:
I made this change.  Personally I find integration tests easier to understand if they don't depend on simapp (less code being run), but I generally agree that relying on simapp in `tests/integration` is OK and we *should* have some tests which do.

Reply:
Using simapp in integration and e2e tests is totally OK!